### ISSUE TITLE
[GLES3] HTML5 exports cause Chrome to lock up for 30 seconds or so

### ISSUE THREAD

**Issue by tavurth** [godotengine/godot#56670]
### Godot version

3.4.2

### System information

OSX

### Issue description

When exporting [my project](https://github.com/tavurth/godot-simple-fps-camera) in GLES3 mode, Chrome will lockup for at least 10 seconds when starting the project every time.

This does not happen with the GLES2 export.

### Steps to reproduce

1. `git clone git@github.com:tavurth/godot-simple-fps-camera.git`
2. Open the project
3. Run the web version from godot

### Minimal reproduction project

https://github.com/tavurth/godot-simple-fps-camera

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/21957. https://github.com/godotengine/godot/pull/53411 may improve the situation (try building `3.x` from source), but it will not help on the first start.

Firefox may fare better with regards to startup times, but GLES3 shader compilation will always take a longer time compared to GLES2.

**Comment by tavurth** [comment]
Hi @Calinou the issue you link to is also closed, I will try to compile the latest 3.x branch, but I feel we should leave at least one of these two issues open until it's resolved?

**Comment by clayjohn** [comment]
Yes, let's leave this open until we have confirmed that shader compilation is the issue and shader caching resolves it.

**Comment by Calinou** [comment]
> I managed to build the OSX 3.x branch, but when trying to build the export templates for HTML5 I run into the following error:

You need to install the Emscripten SDK and load it in your environment by running `./emsdk_env.sh activate`.

**Comment by tavurth** [comment]
Thank you! I will try that now

**Comment by tavurth** [comment]
I can confirm this issue still persists with `shader caching mode` set to `Asynchronous + Cache` in the `GLES 3` options

**Comment by tavurth** [comment]
> Pretty much all hardware still supports WebGL1 significantly better than WebGL2. We were spending too much development effort creating hacks to appease WebGL2 when WebGL1 just works better and is faster anyway.

From the other thread ^

Seems that this is unlikely to be fixed soon, perhaps we can re-open the other issue in place of this one as there is helpful testing which has been listed there?

**Comment by Calinou** [comment]
> Seems that this is unlikely to be fixed soon, perhaps we can re-open the other issue in place of this one as there is helpful testing which has been listed there?

Both issues are unlikely to be fixed soon, so it doesn't really matter which issue is left open by now (or if both are closed).

**Comment by tavurth** [comment]
I see this a lot in public projects now, but I don't remember it being a problem before.

Since this effectively makes my entire browser unusable (Chrome 100.0.4896.20), I think it can be pretty bad for Godot PR.

I would suggest that until this issue is fixed, we limit Web exports to GLES2 to prevent people from pushing more bad PR material about the Godot engine online.

**Comment by Calinou** [comment]
> Alternately, if this issue has been introduced more recently perhaps we can bisect the issue and figure out where it went wrong?

This has always been an issue since Godot 3.0's release. It's just that web browsers are bad at compiling complex WebGL 2.0 shaders. There's nothing we can do about it.

**Comment by tavurth** [comment]
> The documentation already recommends using GLES2 for projects that are planned to be exported to HTML5.

Seems like most people don't read the docs so deeply, and it's a shame to see projects run foul of this. 
In the last month or these two stood out for me:

https://github.com/jayaarrgh/BookOfShaders-Godot/issues/7

https://anima.ceceppa.me/demo/

> I wouldn't prevent exporting GLES3 projects to HTML5, as it does work for some specialized use cases

I notice that Godot has come a long way in implementing warning flags on the nodes when they are applied in the wrong areas. (`_get_configuration_warning`). 

I think this issue might be a great way to push for more of the same, and rather have a large similar warning when people try to export GLES3 for HTML5. What do you think of this idea?

**Comment by tavurth** [comment]
Holy moly, Godot 4.x has just completely broken web exports?

I've been trying to export a game for WebGL but it just hangs on loading? I've noticed the same for other examples as well.

Chrome `119.0.6045.159`

<img width="265" alt="Screenshot 2023-11-28 at 22 21 01" src="https://github.com/godotengine/godot/assets/100964/18948669-9280-4764-82b6-28cbbac68bdb">

**Comment by tavurth** [comment]
@Zireael07 I mean it still causes my Chrome to lockup, it's just more than 30 seconds now 😓 

Actually have to kill the whole browser now whenever a Godot HTML5 game or tool launches.

This tool I've built has zero custom shader usage.

**Comment by Zireael07** [comment]
No, it's not mislabeled, as this issue was opened for a specific Godot version at that time.
You're using 4.x, which is an entirely different beast from 3.4.2

**Comment by akien-mga** [comment]
@tavurth Your issue is #70691. It's a problem specific to Chrome on macOS.

**Comment by tavurth** [comment]
Ok thank you both.

I guess the similarity between both issues is that the shaders requested to be compiled are huge in size?

Is there some way in either godot version (3.x or 4.x) to disable shaders that one doesn't need?

**Comment by tavurth** [comment]
> Yours is a separate issue - this one has been fixed for a year or so

Also anyone still on 3.x can confirm this? Otherwise I guess we would be good to close @Calinou?
